packages/subiquity_client/lib/subiquity_server.dart

```diff
@@ -41,13 +41,15 @@ abstract class SubiquityServer {
       : p.join(Directory.current.path, 'test/socket');
 
   // Optional environment variables for the server process.
-  Map<String, String> _getEnvironment(String subiquityPath) {
+  Map<String, String> _getEnvironment(String? subiquityPath) {
     // prefer local curtin and probert python modules that are pinned to the
     // correct versions
     final pythonPath = (Platform.environment['PYTHONPATH'] ?? '').split(':');
-    pythonPath.add(subiquityPath);
-    pythonPath.add(p.join(subiquityPath, 'curtin'));
-    pythonPath.add(p.join(subiquityPath, 'probert'));
+    if (subiquityPath != null) {
```

Comment:
what about having the directory exists check here instead?

```diff
diff --git a/packages/subiquity_client/lib/subiquity_server.dart b/packages/subiquity_client/lib/subiquity_server.dart
index b862748..a56c887 100644
--- a/packages/subiquity_client/lib/subiquity_server.dart
+++ b/packages/subiquity_client/lib/subiquity_server.dart
@@ -45,9 +45,11 @@ abstract class SubiquityServer {
     // prefer local curtin and probert python modules that are pinned to the
     // correct versions
     final pythonPath = (Platform.environment['PYTHONPATH'] ?? '').split(':');
-    pythonPath.add(subiquityPath);
-    pythonPath.add(p.join(subiquityPath, 'curtin'));
-    pythonPath.add(p.join(subiquityPath, 'probert'));
+    if (Directory(subiquityPath).existsSync()) {
+      pythonPath.add(subiquityPath);
+      pythonPath.add(p.join(subiquityPath, 'curtin'));
+      pythonPath.add(p.join(subiquityPath, 'probert'));
+    }
     return {'PYTHONPATH': pythonPath.join(':')};
   }
 
```

Reply:
Yes, but we will still need to have the null checking here, won’t we? (same issue due to the analyzer)

Reply:
not if we leave out the subiquityPath changes below and keep it non-nullable?